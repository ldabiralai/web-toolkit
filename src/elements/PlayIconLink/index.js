import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { actionOneDarkBase, coreLightMinus1, coreLightBase } from '../../colors';
import { fontFamilies } from '../../typography';
import PlayIcon from '../PlayIcon';
import Link from '../Link';

export const DivPlayIconBg = styled.div`
  display: inline-block;
  vertical-align: top;
  background-color: ${props => props.bgColorIcon};
  height: 100%;
  width: 48px;
  text-align: center;
`;

export const DivTextBg = styled.div`
  display: inline-block;
  vertical-align: top;
  background-color: ${props => props.bgColorText};
  color: ${coreLightBase};
  height: 100%;
  padding: 0 16px;
  text-align: center;
  font-family: ${fontFamilies.interUi};
  font-size: 10px;
  letter-spacing: 1px;
  text-transform: uppercase;
`;

const StyledPlayIcon = styled(PlayIcon)`
  display: inline-block;
  width: 38%;
  vertical-align: middle;
`;

const PlayIconLink = styled(({ children, bgColorIcon, bgColorText, ...props }) => {
  const Wrapper = props.href ? Link : styled.div();
  return (
    <Wrapper {...props}>
      <DivPlayIconBg bgColorIcon={bgColorIcon}>
        <StyledPlayIcon isRounded={false} />
      </DivPlayIconBg>
      <DivTextBg bgColorText={bgColorText}>{children}</DivTextBg>
    </Wrapper>
  );
})`
  position: absolute;
  color: ${coreLightMinus1};
  white-space: nowrap;
  height: 48px;
  line-height: 48px;
`;

PlayIconLink.propTypes = {
  href: PropTypes.string,
  target: PropTypes.string,
  bgColorIcon: PropTypes.string,
  bgColorText: PropTypes.string,
};

PlayIconLink.defaultProps = {
  bgColorIcon: actionOneDarkBase,
  bgColorText: '#3c46dc',
};

export default PlayIconLink;
