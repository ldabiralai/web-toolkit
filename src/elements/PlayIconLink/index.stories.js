import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, color } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { PlayIconLink } from '../..';

const iconsStories = storiesOf('Elements|PlayIconLink', module);

iconsStories.add(
  'default',
  withInfo()(() => (
    <PlayIconLink
      bgColorIcon={color('BgColorIcon', '#4652ff')}
      bgColorText={color('bgColorText', '#3c46dc')}
      href={text('href', 'http://eurosport.fr')}
      target={text('target', '')}
    >
      {text('Link Text', 'Replay ')}
    </PlayIconLink>
  ))
);
