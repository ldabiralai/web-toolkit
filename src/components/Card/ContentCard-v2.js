import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import LinesEllipsis from 'react-lines-ellipsis';
import LinesEllipsisResponsive from 'react-lines-ellipsis/lib/responsiveHOC';
import { coreLightMinus1, coreNeutral3, coreNeutral10, coreLightBase, coreNeutral4 } from '../../colors';
import * as breakpoints from '../../breakpoints';
import { fontFamilies, H2 } from '../../typography';
import Link from '../../elements/Link';
import PlayIconLink from '../../elements/PlayIconLink';

const ResponsiveEllipsis = LinesEllipsisResponsive()(LinesEllipsis);

const LinkCard = styled(Link)`
  overflow: hidden;
  position: relative;
  display: block;
`;

const DivImageContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: ${coreNeutral10};
`;

const SpanBottomOverlay = styled.span`
  position: absolute;
  z-index: 3;
  width: 100%;
  height: 100%;
  left: -10%;
  background-color: ${coreNeutral10};
  top: calc(100% - 65px);
  transform-origin: top left;
  transform: scale(1.3) rotate(-2.6deg);
`;

export const Image = styled.div`
  width: 100%;
  z-index: 1;
  height: 100%;
  background-image: url(${props => props.src});
  background-size: cover;
  background-position: center;
`;

const SpanImageOverlay = styled.span`
  position: absolute;
  z-index: 2;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  opacity: 0.25;
  background: radial-gradient(
    circle,
    rgba(4, 5, 15, 0) 0%,
    rgba(4, 5, 15, 0.06) 5.72%,
    rgba(4, 5, 15, 0.12) 12.1%,
    rgba(4, 5, 15, 0.19) 19.02%,
    rgba(4, 5, 15, 0.26) 26.38%,
    rgba(4, 5, 15, 0.34) 34.07%,
    rgba(4, 5, 15, 0.42) 41.98%,
    rgba(4, 5, 15, 0.5) 50%,
    rgba(4, 5, 15, 0.58) 58.02%,
    rgba(4, 5, 15, 0.66) 65.93%,
    rgba(4, 5, 15, 0.74) 73.62%,
    rgba(4, 5, 15, 0.81) 80.98%,
    rgba(4, 5, 15, 0.88) 87.9%,
    rgba(4, 5, 15, 0.94) 94.28%,
    #04050f 100%
  );
`;

const DivCardContent = styled.div`
  position: relative;
  z-index: 5;
  height: 100%;
`;

const DivTopic = styled.div`
  text-decoration: none;
  display: inline-block;
  font-family: ${fontFamilies.interUi};
  text-transform: uppercase;
  letter-spacing: 1px;
  font-size: 12px;
  line-height: 18px;
`;

const Title = styled(H2)`
  color: ${coreLightMinus1};
  font-family: ${fontFamilies.alphaHeadline};
  font-weight: normal;
`;

const DivDescription = styled.div`
  font-family: ${fontFamilies.interUi};
  font-size: 12px;
  letter-spacing: 1px;
  line-height: 18px;
`;

const DivTextWrapper = styled.div`
  position: absolute;
`;

export const SpanTimeLabel = styled.span`
  position: absolute;
  color: ${coreNeutral4};
  font-family: ${fontFamilies.interUi};
  font-size: 12px;
  letter-spacing: 1px;
`;

const mobileStyles = `
  height: 387px;

  ${SpanBottomOverlay}, ${Image} {
    height: 50%;
  }

  ${DivTopic} {
    margin: 0 24px;
    color: ${coreNeutral3};
  }

  ${Title} {
    color: ${coreLightBase};
    margin: 11.5px 24px 0 24px;
    font-family: ${fontFamilies.interUi};
    font-size: 16px;
    line-height: 24px;
  }

  ${DivDescription} {
    margin: 9px 24px 0 24px;
    color: ${coreNeutral3};
  }

  ${SpanTimeLabel} {
    bottom: 24px;
    left: 24px;
  }

  ${PlayIconLink} {
    bottom: 190px;
    right: 0;
  }

  ${SpanBottomOverlay} {
    top: calc(100% - 200px);
  }

  ${DivTextWrapper} {
    top: 211px;
    bottom: auto;
  }
`;

const tabletStyles = `
  height: 352px;
  
  ${DivTextWrapper} {
    top: 185px;
  }
`;

const desktopStyles = `
  height: 378px;

  ${SpanBottomOverlay}, ${Image} {
    height: 86%;
  }

  ${DivTopic} {
    margin: 0 42px;
    color: ${coreLightBase};
  }

  ${Title} {
    font-size: 28px;
    line-height: 36px;
    margin: 14px 40px 0 40px;

    ${breakpoints.large(`
      font-size: 28px;
      line-height: 36px;
    `)};
  }

  ${DivDescription} {
    margin: 16px 42px 0 42px;
    color: ${coreLightBase};
    text-transform: uppercase;
  }

  ${PlayIconLink} {
    bottom: 71px;
    right: 0;
  }

  ${SpanBottomOverlay} {
    top: calc(100% - 57px);
  }

  ${DivTextWrapper} {
    top: auto;
    bottom: 99px;
  }

  ${SpanTimeLabel} {
    bottom: 28px;
    left: 42px;
  }
`;

const BaseCard = styled(props => (
  <LinkCard className={props.className} href={props.link} data-test="contentCardv2">
    <DivImageContainer>
      {props.image && <Image src={props.image} alt={props.title} />}
      <SpanImageOverlay />
      <SpanBottomOverlay />
    </DivImageContainer>
    <DivCardContent>
      {props.labelPlayButton && <PlayIconLink>{props.labelPlayButton}</PlayIconLink>}
      <DivTextWrapper>
        <DivTopic>{props.topic}</DivTopic>
        <Title>
          <ResponsiveEllipsis text={props.title} maxLine={3} ellipsis="..." trimRight basedOn="letters" />
        </Title>
        <DivDescription>{props.description}</DivDescription>
      </DivTextWrapper>

      {props.children}
    </DivCardContent>
  </LinkCard>
))`
  text-decoration: none;
`;

BaseCard.propTypes = {
  image: PropTypes.string.isRequired,
  topic: PropTypes.string,
  link: PropTypes.string,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  labelPlayButton: PropTypes.string,
};

BaseCard.defaultProps = {
  image: null,
  topic: '',
  title: '',
  description: '',
};

export const CardBig = styled(BaseCard)`
  ${mobileStyles}

  ${breakpoints.medium(`
    ${desktopStyles}
  `)};
`;

export const CardSmall = styled(BaseCard)`
  ${mobileStyles}

  ${breakpoints.medium(`
    ${tabletStyles}
  `)};

  ${breakpoints.large(`
    height: 378px;

    ${DivTextWrapper} {
      top: 211px;
    }
  `)};
`;
