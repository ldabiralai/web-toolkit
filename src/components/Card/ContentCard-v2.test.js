import React from 'react';
import { shallow } from 'enzyme';
import PlayIconLink from '../../elements/PlayIconLink';

import { CardBig, CardSmall, Image } from './ContentCard-v2';

const cardData = {
  image: 'https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800',
  topic: 'Australian Open',
  link: 'https://www.eurosport.fr',
  title: 'Re-Play of the Day: The craziest moments',
  description: 'Roland-garros',
  labelPlayButton: 'Replay',
};

describe('Content card v2', () => {
  it('renders Card with Play Button', () => {
    expect(shallow(<CardBig {...cardData} />)).toMatchSnapshot();
    expect(shallow(<CardSmall {...cardData} />)).toMatchSnapshot();
  });

  it('renders Card without Play Button', () => {
    expect(
      shallow(<CardBig {...cardData} labelPlayButton={undefined} />)
        .dive()
        .find(PlayIconLink).length
    ).toBe(0);
    expect(
      shallow(<CardSmall {...cardData} labelPlayButton={undefined} />)
        .dive()
        .find(PlayIconLink).length
    ).toBe(0);
  });

  it('should render image', () => {
    const wrapper = shallow(<CardBig {...cardData} />);
    expect(wrapper.dive().find(Image).length).toEqual(1);
  });

  it('should not render image tag when no image provided', () => {
    const wrapper = shallow(<CardBig {...cardData} image="" />);
    expect(wrapper.dive().find(Image).length).toBe(0);
  });
});
