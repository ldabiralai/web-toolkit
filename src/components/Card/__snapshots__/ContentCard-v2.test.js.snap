// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content card v2 renders Card with Play Button 1`] = `
<Component
  className="css-orqg7e e129r5mb12"
  description="Roland-garros"
  image="https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800"
  labelPlayButton="Replay"
  link="https://www.eurosport.fr"
  title="Re-Play of the Day: The craziest moments"
  topic="Australian Open"
/>
`;

exports[`Content card v2 renders Card with Play Button 2`] = `
<Component
  className="css-va6jq4 e129r5mb13"
  description="Roland-garros"
  image="https://i.eurosport.com/2019/04/21/2570541-53309530-2560-1440.jpg?w=800"
  labelPlayButton="Replay"
  link="https://www.eurosport.fr"
  title="Re-Play of the Day: The craziest moments"
  topic="Australian Open"
/>
`;
